amr-wind/physics/ForestDrag.cpp

```diff
@@ -0,0 +1,140 @@
+#include "amr-wind/physics/ForestDrag.H"
+#include "amr-wind/CFDSim.H"
+#include "AMReX_iMultiFab.H"
+#include "AMReX_MultiFabUtil.H"
+#include "AMReX_ParmParse.H"
+#include "AMReX_ParReduce.H"
+#include "amr-wind/utilities/trig_ops.H"
+#include "amr-wind/utilities/IOManager.H"
+
+namespace amr_wind::forestdrag {
+
+namespace {} // namespace
+
+ForestDrag::ForestDrag(CFDSim& sim)
+    : m_sim(sim)
+    , m_repo(sim.repo())
+    , m_mesh(sim.mesh())
+    , m_velocity(sim.repo().get_field("velocity"))
+    , m_forest_drag(sim.repo().declare_field("forest_drag", 1, 1, 1))
+    , m_forest_blank(sim.repo().declare_field("forest_blank", 1, 1, 1))
+{
+    std::string forestfile("forest.amrwind");
+    amrex::ParmParse pp(identifier());
+    pp.query("forest_file", forestfile);
+    std::ifstream file(forestfile, std::ios::in);
+    if (!file.good()) {
+        amrex::Abort("Cannot find file " +forestfile);
+    }
+    //! TreeType xc yc height diameter cd lai laimax
+    amrex::Real value1, value2, value3, value4, value5, value6, value7, value8;
+    while (file >> value1 >> value2 >> value3 >> value4 >> value5 >> value6 >>
+           value7 >> value8) {
+        Forest f;
+        f.type_forest = value1;
+        f.x_forest = value2;
+        f.y_forest = value3;
+        f.height_forest = value4;
+        f.diameter_forest = value5;
+        f.cd_forest = value6;
+        f.lai_forest = value7;
+        f.laimax_forest = value8;
+        m_forests.push_back(f);
+    }
+    file.close();
+    m_d_forests.resize(m_forests.size());
+    m_sim.io_manager().register_output_var("forest_drag");
```

Comment:
If they are in the chk file then they have already been written and we do not worry about it. The other option is if more forests are added in restart this may be required to be updated. What do you think is best course of action?

Reply:
I am not sure I follow. If you expect these to change after a restart or if you ever want to tag on them then in general you should follow what I did for TerrainDrag. Move this stuff to initialize_fields.